Content.Server/Store/Conditions/StoreJobCondition.cs

```diff
@@ -0,0 +1,61 @@
+using Content.Server.Mind.Components;
+using Content.Server.Roles;
+using Content.Shared.Roles;
+using Content.Shared.Store;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype.Set;
+
+namespace Content.Server.Store.Conditions;
+
+/// <summary>
+/// Allows a store entry to be filtered out based on the user's job.
+/// Supports both blacklists and whitelists
+/// </summary>
+public sealed class StoreJobCondition : ListingCondition
```

Comment:
Fields and functions of this need doc comments. Remember `<inheritdoc/>` exists for overrides.